### ISSUE TITLE
KeyboardHelp content should be specific to each Screen

### ISSUE THREAD

**Issue by zepumph** [phetsims/joist#668]
Over in https://github.com/phetsims/ratio-and-proportion/issues/82 we have a new requirement for keyboard help content which involves screen-specific content. After a length conversation with @jessegreenberg, we decided that the most ideal solution would be to move the `keyboardHelpNode` option from Sim to Screen. We also want to add an option to Sim.js to opt into having a keyboard help dialog for the whole sim. If you opt in to a keyboard help dialog, but not all screens supply a keyboardHelpNode, then it will just be blank in the dialog. 

We wanted to be as flexible as possible, as we know there are cases like CCK that use alt input for the delete key, but don't opt into a keyboard help dialog. We also wanted to be as explicit as possible, which is why we decided to add a new option to Sim, perhaps called `keyboardHelpDialog:true`. This is will also be a central place, on the Sim level, to describe how keyboard help content can be provided to a phetsim.

Steps for this issue:
1. Move keyboardHelpNode to Screen
3. Add Sim.keyboardHelpDialog boolean
2. refactor logic in KeyboardHelpDialog to support swapping out content based on screen
4. Refactor all keyboard help contents from Sim options to Screen.
5. Make sure documentation is updated about how to add keyboardHelp content to a sim
6. Have someone review this joist change.

@jessegreenberg please add anything else inline if you'd like.

**Comment by zepumph** [comment]
Everything has been done. @jessegreenberg please review. From here I can add screen-specific content to RaP over in https://github.com/phetsims/ratio-and-proportion/issues/82.

**Comment by zepumph** [comment]
Oh, one more piece, @terracoda, I'm not sure that this part of the design was thought about, but since we now base keyboard help content on each screen, the Home Screen gets needs its own. 

I think that the homescreen keyboard dialog will need to pass review by designers, I just made it the basic actions, so every Home Screen keyboard help content looks like this;

![image](https://user-images.githubusercontent.com/6856943/95141332-55364c80-071d-11eb-8bbe-31e1d44c5389.png)

Adding @terracoda to make sure that this is alright for the homescreen dialog.

**Comment by terracoda** [comment]
@zepumph and @jessegreenberg , I think this is great. 

I had thought about the Home Screen would need a Keyboard Help dialog, but I had not yet thought about what content would go in it. The Basic Actions make good sense. The home screen mainly has buttons. The only exception being the menu items in the PhET Menu, where I think you need to use the arrows keys.

I've been thinking that it would be a good idea (and likely simplify the code) if we add arrowing through items in a group as a permanent member of basic actions. I can't think of a good reason not to just explain group navigation, and items in a group all the time.

**Comment by zepumph** [comment]
Added group content above, and created https://github.com/phetsims/scenery-phet/issues/630 for the general refactor. Thanks for the quick review @terracoda. Over to @jessegreenberg.

**Comment by terracoda** [comment]
![Screen Shot 2020-11-26 at 10 49 12 AM](https://user-images.githubusercontent.com/1902116/100361707-101fef80-2fd5-11eb-835d-55558f63b6f4.png)

**Comment by jessegreenberg** [comment]
The options as described in https://github.com/phetsims/joist/issues/668#issue-711447323 look good and are working well, thanks!
